syntax = "proto3";

service CentralRegistry{
    rpc Register(ServerInfo) returns (RegisterResponse);
    rpc MapKey(KeyRequest) returns (MappingResponse);
    rpc Terminate(TerminateRequest) returns (TerminationResponse);
}

message ServerInfo {
    string server_id = 1;
    repeated int32 keys = 2;
}

message KeyRequest {
    int32 key = 1;
}

message RegisterResponse {
    int32 processed_keys = 1;
}

message MappingRequest {
  int32 key = 1;
}

message MappingResponse {
  string server_id = 1;
}


message TerminateRequest {}


message TerminationResponse {
  int32 processed_keys = 1;
}